sos/upload/__init__.py

```diff
@@ -0,0 +1,161 @@
+            """
+
+    arg_defaults = {
+        'upload_url': None,
+        'upload_method': 'auto',
+        'upload_no_ssl_verify': False,
+        'upload_protocol': 'auto',
+        'upload_file': '',
+        'case_id': '',
+        'upload_directory': None,
+    }
+
+    def __init__(self, parser, args, cmdline):
+        # we are running `sos upload` directly
+        # To Do: Work pending on hooking SoSReport or SoSCollector
+        # to this subsystem
+        super().__init__(parser, args, cmdline)
+
+        # add manifest section for upload
+        self.manifest.components.add_section('upload')
+
+    @classmethod
+    def add_parser_options(cls, parser):
+        parser.usage = 'sos upload FILE [options]'
+        upload_grp = parser.add_argument_group(
+            'Upload Options',
+            'These options control how upload manages files'
+            )
+        upload_grp.add_argument("upload_file", metavar="FILE",
+                                help="The file or archive to upload")
+        upload_grp.add_argument("--case-id", action="store", dest="case_id",
+                                help="specify case identifier")
+        upload_grp.add_argument("--upload-url", default=None,
+                                help="Upload the archive to specified server")
+        upload_grp.add_argument("--upload-user", default=None,
+                                help="Username to authenticate with")
+        upload_grp.add_argument("--upload-pass", default=None,
+                                help="Password to authenticate with")
+        upload_grp.add_argument("--upload-directory", action="store",
+                                dest="upload_directory",
+                                help="Specify upload directory for archive")
+        upload_grp.add_argument("--upload-method", default='auto',
+                                choices=['auto', 'put', 'post'],
+                                help="HTTP method to use for uploading")
+        upload_grp.add_argument("--upload-protocol", default='auto',
+                                choices=['auto', 'https', 'ftp', 'sftp'],
+                                help="Manually specify the upload protocol")
+        upload_grp.add_argument("--upload-no-ssl-verify", default=False,
+                                action='store_true',
+                                help="Disable SSL verification for upload url")
+
+    @classmethod
+    def display_help(cls, section):
+        section.set_title('SoS Upload Detailed Help')
+
+        section.add_text(
+            'The upload command is designed to upload already existing '
+            'sos reports, as well as other files like logs and vmcores '
+            'to a distribution specific location.'
+        )
+
+    def _fmt_msg(self, msg):
+        width = 80
+        _fmt = ''
+        for line in msg.splitlines():
+            _fmt = _fmt + fill(line, width, replace_whitespace=False) + '\n'
+        return _fmt
```

Comment:
We declare this method on many places / in most of components. Isn't it worth declaring in `SosComponent` directly?

Reply:
I would prefer having extra commit within this PR, but I am ok with a new issue/PR.